File: lab3/task3.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() { 
    setlocale(LC_ALL, ".UTF8");
    cout << "Enter a, b: \n";
    int a, b; 
    cin >> a >> b; 
    if (b == 1) { 
        cout << "infinity" << endl; 
        return 0; 
    }
    if (a > 10 || a < 1 || b > 10 || b < 1) {
        cout << "error";
        return 1;
    }
    double sum = 0.0; 
    for (int n = 1; n <= 100; n++) { 
        sum += (pow (n,a) / pow(b,n)); 
    }
    int min_num, min_den; 
    double min_dif = 999; 
    for (double num = 1; num <= 1000; num++) 
    {
        for (double den = 1; den <= 1000; den++) {
            long double dif = abs(sum - (num / den)); 
            if (min_dif > dif) { 
                min_dif = dif; 
                min_num = num; 
                min_den = den; 
            }
        }
    }
    cout << "Sum: " << min_num << "/" << min_den; 
    return 0; 
} 

File: lab3/task3.rs
use std::io::*;
use std::process;

fn main() {
    print!("Enter a, b: ");
    io::stdout().flush().unwrap();
    let mut input = String::new();
    io::stdin().read_line(&mut input).expect("Input error");

    let parts: Vec<&str> = input.trim().split_whitespace().collect();
    if parts.len() < 2 {
        println!("error");
        exit(1);
    }

    let a: i32 = parts[0].parse().expect("Wrong a");
    let b: i32 = parts[1].parse().expect("Wrong b");
    if b == 1 {
        println!("infinity");
        return;
    }

    if a < 1 || a > 10 || b < 1 || b > 10 {
        println!("error");
        exit(1);
    }

    let mut sum = 0.0_f64;
    for n in 1..=100 {
        sum += (n as f64).powi(a) / (b as f64).powi(n);
    }
    let mut min_num = 1_i32;
    let mut min_den = 1_i32;
    let mut min_dif = 999.0_f64;

    for num in 1..=1000 {
        for den in 1..=1000 {
            let dif = (sum - (num as f64 / den as f64)).abs();
            if dif < min_dif {
                min_dif = dif;
                min_num = num;
                min_den = den;
            }
        }
    }
    println!("Sum: {}/{}", min_num, min_den);
}